pkg/controller/controllercmd/builder.go

```diff
@@ -70,6 +73,21 @@ func (b *ControllerBuilder) WithFileObserver(reactorFunc func(file string, actio
 	return b
 }
 
+func (b *ControllerBuilder) WithEventRecorder() *ControllerBuilder {
+	clientConfig, err := b.getClientConfig()
+	if err != nil {
+		panic(fmt.Sprintf("error getting client config: %v", err))
+	}
+	kubeClient := kubernetes.NewForConfigOrDie(clientConfig)
+	replicaSetRef, err := events.GetReplicaSetOwnerReference(kubeClient.AppsV1().ReplicaSets(b.componentName))
```

Comment:
I think we should rename that `componentName` to `targetNamespaceName` or something alike, currently it's confusing.

Reply:
These probably need to be separate things?

Reply:
the componentName is currently set to namespace name as we are trying to match those (remember @deads2k talking about us moving namespaces? ;-)

I can add targetNamespace as argument to this func and we can set it to componentName and others to whatever they need :-)